### PR TITLE
add token subcommand

### LINKED ISSUE #11: CLI option for setting token
As the title says: have a subcommand for setting the token. This can be useful in situations where the file structure is _not_ generated by the setup command

subcommand name suggestions:
```
cargo aoc token <>
cargo aoc set-token <>
cargo aoc session <>
cargo aoc session-token <>
```

### PR DESCRIPTION
**Pull request by sivertjoe**
Changes: ⭐ 
* add token subcommand with
    * get token option
    * set token option

This is useful in cases where the filestructure is not generated by the setup command
fixes #11

### CODE REVIEW

File: src/main.rs [hunk 1]
```diff
@@ -18,7 +19,25 @@ async fn main() -> Result<(), Box<dyn std::error::Error>>
...
+                        .short('s')
+                        .long("set")
+                        .exclusive(true)
+                        .help("Set the session token"),
+                    Arg::new("get")
+                        .short('g')
+                        .long("get")
+                        .exclusive(true)
+                        .num_args(0)
+                        .help("Print the current session token, if any"),
```
Comment (seblj) [inline src/main.rs:38]: :heart_eyes: We should add more help messages for the other arguments in a new PR ofc

File: src/token.rs [hunk 2]
```diff
@@ -0,0 +1,27 @@
+use clap::ArgMatches;
+
+use crate::util::file::*;
+
+pub async fn token(matches: &ArgMatches)
+{
+    if let Some(token) = matches.get_one::<String>("set")
+    {
+        let mut path = cargo_path(std::env::current_dir().unwrap()).await.unwrap();
```
Comment (seblj) [inline src/token.rs:9]: Hmmm, should we do the current_dir? :thinking: Would this fail if they are in say `year_2022/src/bin/day_01/` and runs `cargo aoc token set xxx`. Then they move to the directory `year_2022` and try to run `cargo aoc run -d 1 -y 22`. Will it find `.env` and the token then?

**Review by seblj — COMMENTED** [review]
Only one small comment, but I am not 100% it is a problem though, but it may be :man_shrugging:

File: src/main.rs (hunk 1 above)
Comment (sivertjoe) [reply src/main.rs:38]: Yeah

File: src/token.rs (hunk 2 above)
Comment (sivertjoe) [reply src/token.rs:9]: This is maybe just the API for my `cargo_path(path)` being a little bit jank. It puts the env file in the cargo_folder (the same folder as Cargo.toml is located). The argument to the function is just the starting search location. Would probably have been best for to just hard-code it in the function

Comment (seblj) [reply src/token.rs:9]: Ofc my bad🤦‍♂️ I didn't think of it finding the cargo root as I was thinking to much of current_dir. This should work!

**Review by seblj — APPROVED** [review]

**Pull request merged**